projects/public-search/src/app/document-detail/item/item.component.ts

```diff
@@ -24,7 +26,30 @@ import { TranslateService } from '@ngx-translate/core';
 export class ItemComponent {
 
   /** Item record */
-  @Input() item: any;
+  private _item: any;
+
+  /** Temporary item type circulation */
+  circulationInformation: string | undefined = undefined;
+
+  /** Set item record */
+  @Input() set item(item: any) {
```

Comment:
interesting syntax.
Why this one instead of `ngOnInit` or `ngOnChanges` ?

Reply:
Item processing during data injection. No need to call the hook. Makes the class lighter.